src/clj/puppetlabs/services/jruby_pool_manager/impl/jruby_internal.clj

```diff
@@ -221,25 +230,18 @@
   return-to-pool
   "Return a borrowed pool instance to its free pool."
   [instance :- jruby-schemas/JRubyInstanceOrPill]
-  (if (jruby-schemas/jruby-instance? instance)
```

Comment:
I think this `if` statement still needs to be here.  When running the `test-503-when-jruby-is-first-to-shutdown` test from puppetserver, I started hitting schema errors from the `get-instance-state-container` function below during shutdown when a `return-to-pool` call was passed a shutdown poison pill.  If the instance passed in is a shutdown poison pill, I don't think we need to return it to the pool anymore.  But we still should avoid trying to manipulate a `:borrow-count` for it, something which only applies for a real jruby-instance.

The lack of a test failure for this also seems to indicate that maybe we're missing a test here.